src/metaplex/token_metadata.ts

```diff
@@ -62,3 +67,33 @@ export const fetchMetadata = async (
     throw err;
   }
 };
+
+// Returns if Metadata account no longer exists.
+// Safe to retry all other errors w/o suffering from infinite loop.
+export const fetchMetadataAcct = async (
+  conn: Connection,
+  metadataAddr: PublicKey,
+): Promise<Metadata | null> => {
+  try {
+    const acct = await conn.getAccountInfo(metadataAddr);
+    if (!acct) {
+      return null;
+    }
+    return deserializeMeta(acct.data);
+  } catch (err: any) {
+    const errType = getMetadataErrType(err);
+    switch (errType) {
+      case MetadataErrType.Missing:
+        console.warn(
+          `metadata acct ${metadataAddr} does not exist, skipping: ${err}`,
+        );
+        return null;
+      case MetadataErrType.Malformed:
+        console.warn(
+          `metadata acct ${metadataAddr} malformed, skipping: ${err}`,
+        );
+        return null;
+    }
+    throw err;
+  }
+};
```

Comment:
This is identical to fetchMetadata above no?

Reply:
lmao, ok i dont need it anymore then. I cant see.
CLosing PR